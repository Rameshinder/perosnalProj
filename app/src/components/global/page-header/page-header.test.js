import React from 'react'
import { shallow } from 'enzyme'

import PageHeader from './index'

jest.mock('@anz/icon/filled/generic/print', () => 'PrintIcon')
jest.mock('@anz/icon/filled/arrows-and-symbols/question-help', () => 'HelpIcon')

describe('PageHeader', () => {
  const app = shallow(<PageHeader />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('Print should render without crashing', () => {
    shallow(<PageHeader />)
  })

  it('PageContainer should render with action bar', () => {
    shallow(<PageHeader />)
  })
})
