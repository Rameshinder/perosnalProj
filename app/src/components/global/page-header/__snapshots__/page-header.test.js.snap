// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader
    help={true}
    print={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.h1
          data-test-id={undefined}
          proLight={true}
        />,
        <styled.div>
          <Print />
          <Help
            link="http://www.anz.com/internet-banking/help/accounts/"
          />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "data-test-id": undefined,
          "proLight": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Print />,
            <Help
              link="http://www.anz.com/internet-banking/help/accounts/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "link": "http://www.anz.com/internet-banking/help/accounts/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h1
            data-test-id={undefined}
            proLight={true}
          />,
          <styled.div>
            <Print />
            <Help
              link="http://www.anz.com/internet-banking/help/accounts/"
            />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "data-test-id": undefined,
            "proLight": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Print />,
              <Help
                link="http://www.anz.com/internet-banking/help/accounts/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "link": "http://www.anz.com/internet-banking/help/accounts/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
